internal/rpc/clientcache.go

```diff
@@ -64,7 +64,7 @@ func (cc *ClientCache) GetGRPC(address string) (*grpc.ClientConn, error) {
 		defer cancel()
 		for {
 			s := c.client.GetState()
-			if s == connectivity.Ready {
+			if s == connectivity.Ready || s == connectivity.Idle {
```

Comment:
In `fetch_matches_test.go:873`, the assertion is `duration < registrationInterval+proposalCollectionInterval`. I have a feeling that this might be a timing issue but I also think adding idle to the valid states it a net good.